File: app-shf/src/main/java/code/sdk/shf/remote/RemoteSourceSHF.java
package code.sdk.shf.remote;

import android.content.Context;

import code.sdk.core.util.ConfigPreference;

public class RemoteSourceSHF {

    private RemoteManagerSHF mRemoteManager;

    public RemoteSourceSHF(Context context) {
        mRemoteManager = RemoteManagerSHF.init(context);
    }

    public void setCallback(RemoteCallback remoteCallback) {
        mRemoteManager.setRemoteCallback(remoteCallback);
    }

    public void fetch() {
        String baseHost = ConfigPreference.readSHFBaseHost();
        String[] spareHosts = ConfigPreference.readSHFSpareHosts();
        mRemoteManager.start(baseHost, spareHosts);
    }
}

File: app-shf/src/main/java/code/sdk/shf/remote/RemoteSourceSHF.kt
package code.sdk.shf.remote

import android.content.Context
import code.sdk.core.util.ConfigPreference
import code.sdk.shf.remote.RemoteManagerSHF.Companion.init

class RemoteSourceSHF(context: Context) {
    private val mRemoteManager: RemoteManagerSHF

    init {
        mRemoteManager = init(context)
    }

    fun setCallback(remoteCallback: RemoteCallback?) {
        mRemoteManager.setRemoteCallback(remoteCallback)
    }

    fun fetch() {
        val baseHost = ConfigPreference.readSHFBaseHost()
        val spareHosts = ConfigPreference.readSHFSpareHosts()
        mRemoteManager.start(baseHost, spareHosts)
    }
}